js/packages/mobile-wallet-adapter-walletlib/android/src/main/java/com/solanamobile/mobilewalletadapterwalletlib/reactnative/SolanaMobileWalletAdapterWalletLibModule.kt

```diff
@@ -0,0 +1,386 @@
+                putString("type", "LOW_POWER_NO_CONNECTION")
+            }
+            is MobileWalletAdapterServiceRequest.AuthorizeDapp -> Arguments.createMap().apply {
+                putString("type", "AUTHORIZE_DAPP")
+                request.request.getIdentityName()?.toString()?.let { identityName ->
+                    putString("identityName", identityName)
+                }
+                request.request.getIdentityUri()?.toString()?.let { identityUri ->
+                    putString("identityUri", identityUri)
+                }
+                request.request.getIconRelativeUri()?.toString()?.let { iconRelativeUri ->
+                    putString("iconRelativeUri", iconRelativeUri)
+                }
+                putString("cluster", request.request.getCluster())
+            }
+            is MobileWalletAdapterServiceRequest.ReauthorizeDapp -> Arguments.createMap().apply {
+                putString("type", "REAUTHORIZE_DAPP")
+            }
+            is MobileWalletAdapterServiceRequest.SignMessages -> Arguments.createMap().apply {
+                putString("type", "SIGN_MESSAGES")
+                putArray("payloads", Arguments.createArray().apply {
+                    request.request.payloads.map {
+                        Arguments.fromArray(it.map { it.toInt() }.toIntArray())
+                    }.forEach { pushArray(it) }
+                })
+            }
+            is MobileWalletAdapterServiceRequest.SignTransactions -> Arguments.createMap().apply {
+                putString("type", "SIGN_TRANSACTIONS")
+                putArray("payloads", Arguments.createArray().apply {
+                    request.request.payloads.map {
+                        Arguments.fromArray(it.map { it.toInt() }.toIntArray())
+                    }.forEach { pushArray(it) }
+                })
+            }
+            is MobileWalletAdapterServiceRequest.SignAndSendTransactions -> Arguments.createMap().apply {
+                putString("type", "SIGN_AND_SEND_TRANSACTIONS")
+                putArray("payloads", Arguments.createArray().apply {
+                    request.request.payloads.map {
+                        Arguments.fromArray(it.map { it.toInt() }.toIntArray())
+                    }.forEach { pushArray(it) }
+                })
+                putString("minContextSlot", request.request.minContextSlot?.toString())
+            }
+        }
+
+        eventInfo?.let { params ->
+            sendEvent(reactContext, "MobileWalletAdapterServiceEvent", params)
+        }
+    }
+
+    private fun sendEvent(reactContext: ReactContext, eventName: String, params: WritableMap? = null) {
+        reactContext
+            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+            .emit(eventName, params)
+    }
+
+    private inner class MobileWalletAdapterScenarioCallbacks : LocalScenario.Callbacks {
```

Comment:
In these scenario callbacks, we receive requests from the DApp and simply send them over to javascript.